### PR TITLE
arvo: add +de-case to speed up +de-beam, prime cache in +settle:va

### PR DESCRIPTION
**Pull request by tacryt-socryp**
Made `+look` in arvo significantly faster. A trace that has the eyre performance improvements merged in already takes about 120ms less per chat message sent. Down from 300ms to 180ms.

Screen Shot 2021-05-26 at 5 18 07 PM

### CODE REVIEW

**Review by joemfb — CHANGES_REQUESTED** [review]

File: pkg/arvo/sys/arvo.hoon [hunk 1]
```diff
@@ -291,12 +291,13 @@
...
 ::
 ++  de-beam
+  ~/  %de-beam
   |=  p=path
   ^-  (unit beam)
   ?.  ?=([@ @ @ *] p)  ~
   ?~  who=(slaw %p i.p)  ~
   ?~  des=?~(i.t.p (some %$) (slaw %tas i.t.p))  ~  :: XX +sym ;~(pose low (easy %$))
-  ?~  ved=(slay i.t.t.p)  ~
+  ?~  ved=(sloy i.t.t.p)  ~
```
[the remark this answers is not in the document]
Comment (tacryt-socryp) [reply pkg/arvo/sys/arvo.hoon:300]: This looks like a good approach to me. I'll break it out into its own `+de-case` arm for clarity.

[the remark this answers is not in the document]
Comment (tacryt-socryp) [reply pkg/arvo/sys/arvo.hoon]: Done and tested. All good, maybe marginally faster.

File: pkg/arvo/sys/arvo.hoon [hunk 2]
```diff
@@ -291,14 +291,23 @@
...
   ?~  who=(slaw %p i.p)  ~
   ?~  des=?~(i.t.p (some %$) (slaw %tas i.t.p))  ~  :: XX +sym ;~(pose low (easy %$))
-  ?~  ved=(slay i.t.t.p)  ~
-  ?.  ?=([%$ case] u.ved)  ~
-  `(unit beam)`[~ [`ship`u.who `desk`u.des `case`p.u.ved] t.t.t.p]
+  ?~  ved=(en-case i.t.t.p)  ~
+  `(unit beam)`[~ [`ship`u.who `desk`u.des `case`u.ved] t.t.t.p]
+::
+++  en-case
+  ~/  %en-case
```
Comment (joemfb) [inline pkg/arvo/sys/arvo.hoon:304]: Please rename to `+de-case`. `+en-case` would be `|=(=case (scot case))`, if you wanted it.

**Review by joemfb — CHANGES_REQUESTED** [review]

File: pkg/arvo/sys/arvo.hoon [hunk 3]
```diff
@@ -291,14 +291,23 @@
...
   |=  p=path
   ^-  (unit beam)
   ?.  ?=([@ @ @ *] p)  ~
   ?~  who=(slaw %p i.p)  ~
   ?~  des=?~(i.t.p (some %$) (slaw %tas i.t.p))  ~  :: XX +sym ;~(pose low (easy %$))
-  ?~  ved=(slay i.t.t.p)  ~
-  ?.  ?=([%$ case] u.ved)  ~
-  `(unit beam)`[~ [`ship`u.who `desk`u.des `case`p.u.ved] t.t.t.p]
+  ?~  ved=(de-case i.t.t.p)  ~
+  `(unit beam)`[~ [`ship`u.who `desk`u.des `case`u.ved] t.t.t.p]
```
Comment (joemfb) [inline pkg/arvo/sys/arvo.hoon:301]: It looks like there are extra casts in this line, it'd be nice to clean it up. I think only `ship` and `desk` should be necessary:

```
`[[`ship`u.who `desk`u.des u.ved] t.t.t.p]
```

File: pkg/arvo/sys/arvo.hoon [hunk 4]
```diff
@@ -291,14 +291,23 @@
...
+::
+++  de-case
+  ~/  %en-case
```
Comment (joemfb) [inline pkg/arvo/sys/arvo.hoon:304]: This hint should be `%de-case` as well.

**Review by joemfb — CHANGES_REQUESTED** [review]
This looks great -- a couple nits:

**Review by joemfb — APPROVED** [review]
LGTM

**Comment by joemfb** [comment]
This PR should target `release/next-sys`

**Pull request merged**